lib/src/file-input/FileInput.tsx

```diff
@@ -66,31 +66,44 @@ const DxcFileInput = React.forwardRef<RefType, FileInputPropsType>(
     ref
   ): JSX.Element => {
     const [isDragging, setIsDragging] = useState(false);
-    const [files, setFiles] = useState([]);
+    const [files, setFiles] = useState<FileData[]>([]);
     const [fileInputId] = useState(`file-input-${uuidv4()}`);
 
     const colorsTheme = useTheme();
     const translatedLabels = useTranslatedLabels();
 
-    const checkFileSize = (file) => {
+    const checkFileSize = (file: File) => {
       if (file.size < minSize) return translatedLabels.fileInput.fileSizeGreaterThanErrorMessage;
       else if (file.size > maxSize) return translatedLabels.fileInput.fileSizeLessThanErrorMessage;
     };
 
-    const getFilesToAdd = async (selectedFiles) => {
+    const isFileIncluded = (file: FileData, fileList: FileData[]) => {
+      const fileListInfo = fileList.map((existingFile) => existingFile.file);
+      return fileListInfo.some(({ name, size, type, lastModified, webkitRelativePath }) =>
+        name === file.file.name &&
+        size === file.file.size &&
+        type === file.file.type &&
+        lastModified === file.file.lastModified &&
+        webkitRelativePath === file.file.webkitRelativePath
+      )
+    }
+
+    const getFilesToAdd = async (selectedFiles: File[]) => {
       const filesToAdd = await Promise.all(selectedFiles.map((selectedFile) => getFilePreview(selectedFile))).then(
-        (previews) =>
+        (previews: string[]) =>
           selectedFiles.map((file, index) => {
             const fileInfo = { file, error: checkFileSize(file), preview: previews[index] };
             return fileInfo;
           })
       );
-      return filesToAdd;
+      return filesToAdd.filter(
+        file => !isFileIncluded(file, files)
+      )
     };
 
     const addFile = async (selectedFiles) => {
+      const filesToAdd = await getFilesToAdd(selectedFiles);
```

Comment:
Yes, I completely missed this code repetition.

Instead of simply adding it to the line 111 I just did some restructuring of the method in order for it to look cleaner and more concise (while obviosly maintaining the expected behavior). 
I did it also because it was unefficient to call `getFilesToAdd(selectedFiles)` at the beginning when it is possibly not being used (in case it was a non-multiple `FileInput`)